### ISSUE TITLE
Migrate on js-debug extension's debugging core

### ISSUE THREAD

**Issue by RedMickey** [microsoft/vscode-react-native#1209]
We are planning to migrate React Native extension debugging to [js-debug](https://github.com/microsoft/vscode-js-debug), since currently used [vscode-chrome-debug-core](https://github.com/microsoft/vscode-chrome-debug-core) package will be deprecated soon. For providing communication between a RN app and `js-debug` extension the special [CDP proxy](https://github.com/microsoft/vscode-cdp-proxy) will be used. It will allow to adapt RN platform specific features on the debugging protocol level.

Below there is the migration roadmap for tracking the progress.

## Migration roadmap

- [x] **[extension]** Implement Debug Configuration Provider descriptor factory to get rid of separated extension and debugger processes 
- [x] **[extension]** Implement launch of CDP proxy 
- [x]  **[proxy]** Implement translation of a target URL to inspector Websocket
- [x] **[extension]** Add listeners on CDP events inside the proxy for standard React Native 
- [x] **[extension]** Add listeners on CDP events inside the proxy for React Native Hermes 
- [x] **[extension]** Implement starting of js-debug extension's debug session 
- [x] **[extension]** Adapt current implementation of pre-attach setup for Android to proxy approach 
- [x] **[extension]** Adapt current implementation of pre-attach setup for iOS to proxy approach
  - [x] Adapt implementation of launch and attach preparations
  - [x] Adapt implementation of handling "Reload App" and "Show DevMenu" React Native commands
- [x] **[extension]** Adapt current implementation of pre-attach setup for Hermes to proxy approach
- [x] **[extension]** Adapt current implementation of pre-attach setup for RNW to proxy approach  
- [x] **[extension]** Adapt multi-root approach
- [x] **[proxy]** Implement sourcemap path resolving 
- [x] **[proxy]** Add generic adapter for custom handling of CDP methods
- [x] **[extension]** Improve the stability of the updated extension
- [x] **[smoke tests]** Update smoke tests to suit the corresponding changes  
- [x] **[CI]** Update CI pipelines to suit the corresponding changes
- [x] **[testing]** Test updated React Native extension to make sure that all debug arguments work well

**Comment by gfrancischini** [comment]
Hi, is there any update or ETA for this? I really hope it improves the debug of async functions.

Keep the good work!

**Comment by RedMickey** [comment]
Hi @gfrancischini, unfortunately I don't have the exact date. You can find our current results in [`dev/js-debug-adaptation`](https://github.com/microsoft/vscode-react-native/tree/dev/js-debug-adaptation) branch. Feel free to try it out and let us know if you faced any problems or bugs.

**Comment by SounD120** [comment]
@gfrancischini The [preview version](https://marketplace.visualstudio.com/items?itemName=msjsdiag.vscode-react-native-preview) of the updated extension has been released, so you can try it out and send us feedback on the updated debugging experience. Thank you.

**Comment by gfrancischini** [comment]
Thank you, I'm giving a try right now

**Comment by gfrancischini** [comment]
Hi @SounD120 I didn't had enough time to start digging on this. I tried to use it on the release data (Jun 3) but it was not working for me. Now that I'm free again I started looking again into the preview extension but it still not working. I'm able to attach the debug to the packager but my app never get loaded (JavaScript does not geto loaded when debugging with this version)
I'm currently using react 0.61 with Hermes and my app is a brownfield app (android native activities + react native screens).
Is there any guidance and minimum requirements for the the new debug system?
